cmd/vsphere-xcopy-volume-populator/internal/powerstore/powerstore.go

```diff
@@ -34,27 +34,21 @@ func (p *PowerstoreClonner) CurrentMappedGroups(targetLUN populator.LUN, mapping
 	}
 
 	mappedHosts := make([]string, 0, len(mappings))
+
 	for _, mapping := range mappings {
 		host, err := p.Client.GetHost(ctx, mapping.HostID)
 		if err != nil {
 			klog.Warningf("Failed to get host info for host ID %s: %s", mapping.HostID, err)
 			continue
 		}
 		mappedHosts = append(mappedHosts, host.Name)
+		if _, ok := mappingContext[ocpRealHostNameKey]; !ok {
+			mappingContext[ocpRealHostNameKey] = host.Name
```

Comment:
this looks like it is going to be overwritten over an over again. why not just return right after that line?